src/Featurizers/Components/DocumentStatisticsEstimator.h

```diff
@@ -86,27 +49,28 @@ class DocumentStatisticsAnnotationData {
     // |  Public Types
     // |
     // ----------------------------------------------------------------------
-    using FrequencyMap                      = std::unordered_map<std::string, std::uint32_t>;
-    using IndexMap                          = FrequencyMap;
+    using FrequencyAndIndexMap              = std::unordered_map<std::string, FrequencyAndIndex>;
 
     // ----------------------------------------------------------------------
     // |
     // |  Public Data
     // |
     // ----------------------------------------------------------------------
-    FrequencyMap const                      TermFrequency;                  /// Words and the number of documents that it appears in
-    IndexMap const                          TermIndex;
+    FrequencyAndIndexMap const              TermFrequencyAndIndex;
     std::uint32_t const                     TotalNumDocuments;
 
     // ----------------------------------------------------------------------
     // |
     // |  Public Methods
     // |
     // ----------------------------------------------------------------------
-    DocumentStatisticsAnnotationData(FrequencyMap termFreq, IndexMap termIndex, std::uint32_t totalNumDocuments);
-    ~DocumentStatisticsAnnotationData(void) = default;
+    DocumentStatisticsAnnotationData(DocumentStatisticsAnnotationData &&other) :
```

Comment:
Separate declaration and definition; also move this standard constructor above this move constructor.

Reply:
fixed